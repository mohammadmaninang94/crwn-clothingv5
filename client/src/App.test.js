import AppContainer from './App.container';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Route } from 'react-router-dom';

import HomePage from './pages/home-page/home-page.component';

let appWithoutUser = null;
let appWithUser = null;
let pathMap = {};

describe('Main Page', () => {
    beforeAll(() => {
        const currentUser = {};
        appWithoutUser = shallow(<AppContainer currentUser={null} />);
        appWithUser = shallow(<AppContainer currentUser={currentUser} />);

        const component = shallow(<AppContainer currentUser={null}/>);
        pathMap = component.find(Route).reduce((pathMap, route) => {
            const routeProps = route.props();
            console.log(routeProps.render());
            pathMap[routeProps.path] = routeProps.component;
            return pathMap;
        }, {});
        console.log(pathMap)
    });

    it('should show Home component for / router (getting array of routes)', () => {

        expect(pathMap['/']).toBe(undefined);
      })

    it('renders AppContainer without currentUser', () => {
        expect(appWithoutUser).toMatchSnapshot();
    });

    it('renders AppContainer with currentUser', () => {
        expect(appWithUser).toMatchSnapshot();
    });
});