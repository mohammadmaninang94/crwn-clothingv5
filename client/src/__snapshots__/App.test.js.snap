// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page renders AppContainer with currentUser 1`] = `ShallowWrapper {}`;

exports[`Main Page renders AppContainer without currentUser 1`] = `ShallowWrapper {}`;
